packages/core/src/actions/execute.ts

```diff
@@ -58,88 +59,172 @@ export const executeTask = async (args: {
-          setStorageBatch.map((action) => action.proof.actionIndex),
-          setStorageBatch.map((action) => action.proof.siblings)
+      // We can return early if there are no actions to execute.
+      if (filtered.length === 0) {
+        logger.info('no actions left to execute')
+        return
+      }
+
+      let executed = 0
+      while (executed < filtered.length) {
+        // Figure out the maximum number of actions that can be executed in a single batch.
+        const batchSize = await findMaxBatchSize(filtered.slice(executed))
+
+        // Pull out the next batch of actions.
+        const batch = filtered.slice(executed, executed + batchSize)
+
+        // Keep 'em notified.
+        logger.info(
+          `executing actions ${executed} to ${executed + batchSize} of ${
+            filtered.length
+          }...`
         )
-      ).wait()
-      currActionsExecuted += setStorageBatch.length
+
+        // Execute the batch.
+        await (
+          await chugSplashManager.executeMultipleActions(
+            batch.map((action) => action.action),
+            batch.map((action) => action.proof.actionIndex),
+            batch.map((action) => action.proof.siblings)
+          )
+        ).wait()
+
+        // Move on to the next batch if necessary.
+        executed += batchSize
+      }
     }
 
-    logger.info(
-      `State variables have been set. Deploying the implementation contracts...`
+    // Find the indices of the first DeployImplementation and SetImpl actions so we know where to
+    // split up our batches. Actions have already been sorted in the order SetStorage,
+    // DeployImplementation, SetImplementation. Although we could execute SetStorage and
+    // DeployImplementation actions together, SetStorage actions can often fit in a single batch
+    // so it's more efficient to execute them separately.
+    const firstDepImpl = bundle.actions.findIndex((action) =>
+      isDeployImplementationAction(fromRawChugSplashAction(action.action))
+    )
+    const firstSetImpl = bundle.actions.findIndex((action) =>
+      isSetImplementationAction(fromRawChugSplashAction(action.action))
     )
 
-    // Execute DeployImplementation actions in series. We execute them one by one since each one
-    // costs significantly more gas than a setStorage action (usually in the millions).
-    for (let i = currActionsExecuted; i < firstSetImplIndex; i++) {
-      const action = bundle.actions[i]
-      await (
-        await chugSplashManager.executeChugSplashAction(
-          action.action,
-          action.proof.actionIndex,
-          action.proof.siblings
-        )
-      ).wait()
-      currActionsExecuted += 1
-      logger.info(
-        `Deployed implementation contract: ${
-          currActionsExecuted - firstDeployImplIndex
-        }/${firstSetImplIndex - firstDeployImplIndex}`
+    // Execute SetStorage actions in batches.
+    logger.info(`executing SetStorage actions`)
```

Comment:
nit: Missing `...` at the end. (To keep things consistent with the next couple log messages

Reply:
Done!